interfaces/tests/socket_transport_int/socket_transport_int.c

```diff
@@ -234,4 +253,160 @@ TEST_FUNCTION(send_and_receive_random_buffer_of_random_byte_succeeds)
+        chaos_knight_test.client_socket_handles[i] = socket_transport_create_client();
+
+        ASSERT_IS_NOT_NULL(chaos_knight_test.client_socket_handles[i]);
+
+    }
+
+    THREAD_HANDLE thread;
+
+    interlocked_exchange(&chaos_knight_test.API_create_result, 0);
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, ThreadAPI_Create(&thread, thread_worker_func, &chaos_knight_test));
+    wait_for_value(&chaos_knight_test.API_create_result, 1);
+
+    for (size_t i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        uint8_t buffer_count = 8;
+
+        SOCKET_BUFFER* send_data;
+        SOCKET_BUFFER* recv_data;
+
+        uint32_t data_size = 64;
+
+        uint32_t total_bytes = make_send_recv_buffer(buffer_count, data_size, &send_data, &recv_data);
+
+        uint32_t bytes_recv;
+        uint32_t bytes_sent;
+        uint32_t flag = SOCKET_SEND_FLAG;
+
+        // Send data back and forth
+        ASSERT_ARE_EQUAL(SOCKET_SEND_RESULT, SOCKET_SEND_OK, socket_transport_send(chaos_knight_test.client_socket_handles[i], send_data, buffer_count, &bytes_sent, flag, NULL));
+        ASSERT_ARE_EQUAL(int, total_bytes, bytes_sent);
+
+        ASSERT_ARE_EQUAL(SOCKET_RECEIVE_RESULT, SOCKET_RECEIVE_OK, socket_transport_receive(chaos_knight_test.incoming_socket_handles[i], recv_data, buffer_count, &bytes_recv, 0, NULL));
+        ASSERT_ARE_EQUAL(int, total_bytes, bytes_recv);
+
+        for (uint32_t inner = 0; inner < buffer_count; inner++)
+        {
+            ASSERT_ARE_EQUAL(int, 0, memcmp(send_data[inner].buffer, recv_data[inner].buffer, send_data[inner].length), "mismatch data from array index %" PRIu32 "", inner);
+
+            free(send_data[inner].buffer);
+            free(recv_data[inner].buffer);
+        }
+        free(send_data);
+        free(recv_data);
+    }
+
+    // server disconnects
+    socket_transport_disconnect(chaos_knight_test.incoming_socket_handles[FAILED_SERVER_NUM]);
+
+    // resend data back and forth with disconnected
+    for (size_t i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        uint8_t buffer_count = 8;
+
+        SOCKET_BUFFER* send_data;
+        SOCKET_BUFFER* recv_data;
+
+        uint32_t data_size = 64;
+
+        uint32_t total_bytes = make_send_recv_buffer(buffer_count, data_size, &send_data, &recv_data);
+
+        uint32_t bytes_recv;
+        uint32_t bytes_sent;
+
+        uint32_t flag = SOCKET_SEND_FLAG;
+
+        SOCKET_SEND_RESULT send_result = socket_transport_send(chaos_knight_test.client_socket_handles[i], send_data, buffer_count, &bytes_sent, flag, NULL);
+        if (send_result == SOCKET_SEND_OK)
+        {
+            ASSERT_ARE_EQUAL(int, total_bytes, bytes_sent);
+        }
+
+        if (i == FAILED_SERVER_NUM)
+        {
+            ASSERT_ARE_EQUAL(SOCKET_RECEIVE_RESULT, SOCKET_RECEIVE_ERROR, socket_transport_receive(chaos_knight_test.incoming_socket_handles[i], recv_data, buffer_count, &bytes_recv, 0, NULL));
+            ASSERT_ARE_NOT_EQUAL(int, total_bytes, bytes_recv);
```

Comment:
>ASSERT_ARE_NOT_EQUAL(int, total_bytes, bytes_recv); [](http://example.com/codeflow?start=12&length=51)

If the above function fail, then this call is invalid and doesn't make any sense.